ZenPacks/training/OpenWeather/dsplugins.py

```diff
@@ -37,47 +39,91 @@ def config_key(cls, datasource, context):
     def params(cls, datasource, context):
         return {
             'api_key': context.zOWeatherAPIKey,
+            'api_version': context.zOWeatherAPIVersion,
             'city_name': context.city_name,
             'country_code': context.country_code,
             'location_name': context.title,
             }
 
+    def camel_case_to_snake(self, name):
```

Comment:
cover this function by unitests. create module util.py and move it. Docstrings as well.